tools/mage/js.go

```diff
@@ -26,326 +26,293 @@ import (
-		if changed || os.IsNotExist(err) {
-			if mg.Verbose() {
-				fmt.Println("Running Webpack for DLL...")
-			}
-			webpack, err := js.webpack()
-			if err != nil {
-				return err
-			}
-			return webpack("--config", "config/webpack.dll.babel.js")
-		}
+	if js.isProductionMode() {
+		fmt.Println("Skipping DLL building (production mode)")
 		return nil
 	}
+
+	ok, err := target.Path(
+		filepath.Join("public", "libs.bundle.js"),
+		"yarn.lock",
+	)
+	if err != nil {
+		return targetError(err)
+	}
+	if !ok {
+		return nil
+	}
+	mg.Deps(js.Deps)
 	if mg.Verbose() {
-		fmt.Println("Skipping DLL module bundling (production mode)")
+		fmt.Println("Running Webpack for DLL")
 	}
-	return nil
+	return js.runWebpack("config/webpack.dll.babel.js")
 }
 
-// Serve builds necessary bundles and serves the console for development.
-func (js Js) Serve() {
-	mg.Deps(js.ServeMain)
+// Build runs the webpack command with the project config.
+func (js Js) Build() error {
+	mg.Deps(js.Deps, js.Translations, js.BackendTranslations, js.BuildDll)
+	if mg.Verbose() {
+		fmt.Println("Running Webpack")
+	}
+	return js.runWebpack("config/webpack.config.babel.js")
 }
 
-// ServeMain runs webpack-dev-server
-func (js Js) ServeMain() error {
-	mg.Deps(js.Translations, js.BackendTranslations, js.BuildDll)
+// Serve runs webpack-dev-server.
+func (js Js) Serve() error {
+	mg.Deps(js.Deps, js.Translations, js.BackendTranslations, js.BuildDll)
 	if mg.Verbose() {
-		fmt.Println("Running Webpack for Main Bundle in watch mode...")
-	}
-	webpackServe, err := js.webpackServe()
-	if err != nil {
-		return err
+		fmt.Println("Running Webpack for Main Bundle in watch mode")
 	}
 	os.Setenv("DEV_SERVER_BUILD", "true")
-	return webpackServe("--config", "config/webpack.config.babel.js", "-w")
+	return js.runYarnCommandV("webpack-dev-server",
+		"--config", "config/webpack.config.babel.js",
+		"-w",
+	)
 }
 
 // Messages extracts the frontend messages via babel.
 func (js Js) Messages() error {
-	changed, err := target.Dir("./.cache/messages", "./pkg/webui/console")
-	if os.IsNotExist(err) || (err == nil && changed) {
-		if mg.Verbose() {
-			fmt.Println("Extracting frontend messages...")
-		}
-		babel, err := js.babel()
-		if err != nil {
-			return err
-		}
-		if err = sh.Rm(".cache/messages"); err != nil {
-			return err
-		}
-		if err = os.MkdirAll("pkg/webui/locales", 0755); err != nil {
-			return err
-		}
-		return babel("pkg/webui")
+	mg.Deps(js.Deps)
+	ok, err := target.Dir(
+		filepath.Join(".cache", "messages"),
+		filepath.Join("pkg", "webui", "console"),
+	)
+	if err != nil {
+		return targetError(err)
 	}
-	return nil
+	if !ok {
+		return nil
+	}
+	if mg.Verbose() {
+		fmt.Println("Extracting frontend messages")
+	}
+	if err = sh.Rm(".cache/messages"); err != nil {
+		return err
+	}
+	if err = os.MkdirAll("pkg/webui/locales", 0755); err != nil {
+		return err
+	}
+	return js.runYarnCommand("babel", "pkg/webui")
```

Comment:
This forces `babel` to output all source to `stdout` (see travis log). 
Consider
```go
err = execYarn(nil, os.Stderr, "babel", "pkg/webui")
  if err != nil {
	return err
}
```